### ISSUE TITLE
perf(losses): total_variation reduces with `sum(dim=(-2, -1))`, a trailing-two-dims reduction that MPS runs 24x slower than CPU; `flatten(-2).sum(-1)` is identical and 4x faster there

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4526]
## Problem

`kornia.losses.total_variation` sums its absolute differences with `.sum(dim=(-2, -1))`. On MPS a reduction over the trailing two dims is a pathological path: `x.sum(dim=(-2,-1))` on (1,3,1024,1024) takes 5.05 ms vs 0.21 ms on the M1 CPU, while `x.sum()` and `x.sum(dim=-1)` on the same tensor take 0.5 ms on MPS (standalone numbers in #4522). The loss is therefore slower on MPS than on CPU at 1024² even though its elementwise part is faster there.

| (1,3,1024,1024) | CPU | MPS |
|---|---:|---:|
| `total_variation` today | 8.5 ms | 11.7 ms |
| same with `.flatten(-2).sum(-1)` | 8.5 ms | **3.0 ms** |

Max abs difference between the two: 0.0.

## Fix

Replace the two `.sum(dim=(-2, -1))` with `.flatten(-2).sum(-1)` (or `.sum(-1).sum(-1)`); the `reduction` argument from #1815 is unaffected.

```python
import torch, kornia.losses as L
t = torch.rand(1, 3, 1024, 1024)
tv = (t[..., :, 1:] - t[..., :, :-1]).abs().flatten(-2).sum(-1) + (t[..., 1:, :] - t[..., :-1, :]).abs().flatten(-2).sum(-1)
print((L.total_variation(t) - tv).abs().max().item())  # 0.0
```

Measured on torch nightly 2.15.0.dev20260913, Apple M1 8 GB, macOS 26.5.1, kornia `633951bd7`, `PYTORCH_ENABLE_MPS_FALLBACK` unset, `torch.utils.benchmark` medians with `mps.synchronize` inside the timed region, fresh process per op. Split out of the MPS-audit tracking issue #4521; the PyTorch-side kernel findings are in #4522.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
